pkg/probe/http/http.go

```diff
@@ -53,7 +53,9 @@ func NewWithTLSConfig(config *tls.Config, followNonLocalRedirects bool) Prober {
 			DisableKeepAlives:  true,
 			Proxy:              http.ProxyURL(nil),
 			DisableCompression: true, // removes Accept-Encoding header
+			DialContext:        probe.ProbeDialer().DialContext,
```

Comment:
Comment that net/http is smart enough to use this if we don't spec the SSL function hook?

Reply:
sure